packages/compass-data-modeling/src/components/relationship-drawer-content.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { useCallback, useMemo, useRef } from 'react';
+import { connect } from 'react-redux';
+import type { DataModelingState } from '../store/reducer';
+import {
+  Button,
+  Combobox,
+  FormFieldContainer,
+  H3,
+  ComboboxOption,
+  Select,
+  Option,
+} from '@mongodb-js/compass-components';
+import {
+  deleteRelationship,
+  getCurrentDiagramFromState,
+  selectFieldsForCurrentModel,
+  selectRelationshipForCurrentModel,
+  updateRelationship,
+} from '../store/diagram';
+import toNS from 'mongodb-ns';
+import type { Relationship } from '../services/data-model-storage';
+import { cloneDeep } from 'lodash';
+
+type RelationshipDrawerContentProps = {
+  relationshipId: string;
+  relationship: Relationship;
+  fields: Record<string, string[][]>;
+  onRelationshipUpdate: (relationship: Relationship) => void;
+  onDeleteRelationshipClick: (rId: string) => void;
+};
+
+type RelationshipFormFields = {
+  localCollection: string;
+  localField: string;
+  foreignCollection: string;
+  foreignField: string;
+  localCardinality: string;
+  foreignCardinality: string;
+};
+
+const FIELD_DIVIDER = '~~##$$##~~';
```

Comment:
There's always a chance someone will have a field like that, but that's probably rare enough?